pennylane/workflow/qnode.py

```diff
@@ -273,9 +273,30 @@ class QNode:
               rule for all supported quantum operation arguments, with finite-difference
               as a fallback.
 
-            * ``"hadamard"``: Use the analytic hadamard gradient test
-              rule for all supported quantum operation arguments. More info is in the documentation
-              :func:`qml.gradients.hadamard_grad <.gradients.hadamard_grad>`.
+            * ``"hadamard"``: Use the standard analytic hadamard gradient test rule for 
+            all supported quantum operation arguments. More info is in the documentation
+            for :func:`qml.gradients.hadamard_grad <.gradients.hadamard_grad>`.
+
+            * ``"reversed-hadamard"``: A variant of ``"hadamard"``, where the role of the 

```

Comment:
At this point, the different modes are accessible via:
```
qml.QNode(f, dev, diff_method="hadamard", gradient_kwargs={"mode": "reversed"})
```
To automatically choose the mode from a string `diff_method`, we will need to update this workflow function:

https://github.com/PennyLaneAI/pennylane/blob/db404c0e0f2a663a00ecfcd4cbc56d4fdc984751/pennylane/workflow/resolution.py#L163

to be able to interpret the various hadamard options. We can do so in this PR, but I'd prefer to do that in a follow on.